### PR TITLE
perf(utils/dom): use passive event listeners where possible

### LINKED ISSUE #1858: Warnings on b-form-select open/close in recent Chrome
**To recreate:**
  Go to https://bootstrap-vue.js.org/play/ in Chrome v66 (I believe v51 or newer) and enter the following template:
```html
  <b-form-select>
      <option>Sample</option>
  </b-form-select>
```
Then open and close the resulting select input several times.

**Expected:**
Developer console free of warnings and errors.

**Actual:**
See the following warning in the JavaScript console: 

> [Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

### PR DESCRIPTION
**Pull request by tmorehouse**
## Description of Pull Request:

Adds detection of event passive support to dom utils.  Normalizes 3rd argument to `addEventListener` and `removeEventListener`.

Passive mode allows the browser (that support passive mode) to optimize how it handles cancellable events that will never be cancelled (such as scroll/mouewheel/touch event handlers that will never call `event.preventDefault()`), improving scroll performance

Now used in `v-b-scrollspy`, `b-carousel`, and `b-img-lazy`

Related issue: #1858 (although it is codemirror that is generating those warnings)

Closes #1858

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)
- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [ ] Documentation update
- [x] Other, please describe: performance

**Does this PR introduce a breaking change?** (check one)
- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**
- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e. `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos", "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**
- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives 
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's description above includes:**
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/carousel/carousel.js [hunk 1]
```diff
@@ -527,14 +529,14 @@ export default {
     }
     // Touch support event handlers for environment
     if (!this.noTouch && hasTouchSupport) /* istanbul ignore next: JSDOM doesn't support touch events */ {
-      // Attach appropriate listeners
+      // Attach appropriate listeners (passsive mode)
       if (hasPointerEvent) {
-        on.pointerdown = this.touchStart
-        on.pointerup = this.touchEnd
+        on['&pointerdown'] = this.touchStart
```
Comment (jackmu95) [inline src/components/carousel/carousel.js]: Why is there a `&` prefixed to the event names?

Comment (tmorehouse) [reply src/components/carousel/carousel.js]: `&` is the render function version of the event modifier `.passive` (https://vuejs.org/v2/guide/render-function.html#Event-amp-Key-Modifiers)

Comment (jackmu95) [reply src/components/carousel/carousel.js]: Thanks for pointing out!

Comment (tmorehouse) [reply src/components/carousel/carousel.js]: No prob!

**Pull request merged**